blocks/library/embed/index.js

```diff
@@ -109,38 +109,31 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 				}
 				const { url } = this.props.attributes;
 				const { setAttributes } = this.props;
-				const apiURL = addQueryArgs( wpApiSettings.root + 'oembed/1.0/proxy', {
-					url: url,
-					_wpnonce: wpApiSettings.nonce,
-				} );
 
 				this.setState( { error: false, fetching: true } );
-				window.fetch( apiURL, {
-					credentials: 'include',
-				} ).then(
-					( response ) => {
+				wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ).then(
+					( obj ) => {
 						if ( this.unmounting ) {
 							return;
 						}
-						response.json().then( ( obj ) => {
-							const { html, provider_name: providerName } = obj;
-							const providerNameSlug = kebabCase( toLower( providerName ) );
-							let { type } = obj;
-
-							if ( includes( html, 'class="wp-embedded-content" data-secret' ) ) {
-								type = 'wp-embed';
-							}
-							if ( html ) {
-								this.setState( { html, type, providerNameSlug } );
-								setAttributes( { type, providerNameSlug } );
-							} else if ( 'photo' === type ) {
-								this.setState( { html: this.getPhotoHtml( obj ), type, providerNameSlug } );
-								setAttributes( { type, providerNameSlug } );
-							} else {
-								this.setState( { error: true } );
-							}
-							this.setState( { fetching: false } );
-						} );
+
+						const { html, provider_name: providerName } = obj;
+						const providerNameSlug = kebabCase( toLower( providerName ) );
+						let { type } = obj;
+
+						if ( includes( html, 'class="wp-embedded-content" data-secret' ) ) {
+							type = 'wp-embed';
+						}
+						if ( html ) {
+							this.setState( { html, type, providerNameSlug } );
+							setAttributes( { type, providerNameSlug } );
+						} else if ( 'photo' === type ) {
+							this.setState( { html: this.getPhotoHtml( obj ), type, providerNameSlug } );
+							setAttributes( { type, providerNameSlug } );
+						} else {
+							this.setState( { error: true } );
+						}
```

Comment:
This breaks the embed blocks here.

With `fetch`, we get the error response from oembed back here as json, so this is where we deal with errors. With `wp.apiRequest`, we don't. So if anyone puts in a URL that we can't embed, we never get out of the `fetching` state and report the error to the user.

Reply:
Good catch @notnownikki It should be fixed now.

Reply:
:+1: You beat me to the fix :) works as expected now!